pkg/controller/multiarchbuildconfig/mirror.go

```diff
@@ -0,0 +1,103 @@
+package multiarchbuildconfig
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	v1 "github.com/openshift/ci-tools/pkg/api/multiarchbuildconfig/v1"
+)
+
+const (
+	// Condition types
+	MirrorImageManifestDone string = "ImageMirrorDone"
+
+	// Condition reasons
+	ImageMirrorErrorReason   string = "ImageMirrorError"
+	ImageMirrorSuccessReason string = "ImageMirrorSuccess"
+)
+
+// handleMirrorImage pushes an image to the locations specified in .spec.output.to. The function requires an image
+// to have been already created and pushed to the local registry.
+func (r *reconciler) handleMirrorImage(srcImage string, mabc *v1.MultiArchBuildConfig) (func(mabcToMutate *v1.MultiArchBuildConfig), bool) {
+	if len(mabc.Spec.Output.To) == 0 {
```

Comment:
This will never happen based on https://github.com/openshift/ci-tools/pull/3676/files#diff-a3312e2e48e4c26f431b062d5e8c3fd8b2c2abe781a6913135f17d2d4620afefR44-R45

Reply:
Yeah, that's right thanks. But still I need to check `if mabc.Spec.Output.To != nil`

Reply:
You don't need to do that, because it will never happen.